core/primitives/src/errors.rs

```diff
@@ -76,15 +76,27 @@ impl std::fmt::Display for RuntimeError {
 
 impl std::error::Error for RuntimeError {}
 
+/// Contexts in which `StorageError::MissingTrieValue` error might occur.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum MissingTrieValueContext {
+    /// A trie iterator found a node hash for which there is no corresponding value in storage.
+    TrieIterator,
+    /// A hash found in a trie had no node while fetching from DB.
+    TriePrefetchingStorage,
+    /// Trie memory recorded storage does not have node of given hash.
+    TrieMemoryPartialStorage,
+    /// Missing value while reading trie node from DB.
+    TrieStorage,
```

Comment:
Those seem to be the same error just coming out of the different implementations of TrieStorage. Can you adjust the comments to reflect that? Maybe something like "Missing trie value when reading from X" for X in [TriePrefetchingStorage, TrieMemoryPartialStorage, TrieStorage]. Please correct me if I'm wrong here.